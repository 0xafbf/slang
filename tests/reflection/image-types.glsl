//TEST(smoke):SIMPLE:-profile ps_4_0 -target reflection-json

// Confirm that we expose GLSL `image` types through reflection

uniform imageBuffer iBuffer;

uniform image2D i2D;

void main()
{}
