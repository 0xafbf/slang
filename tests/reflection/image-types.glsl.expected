result code = 0
standard error = {
}
standard output = {
{
    "parameters": [
        {
            "name": "iBuffer",
            "binding": {"kind": "descriptorTableSlot", "index": 0},
            "type": {
                "kind": "resource",
                "baseShape": "textureBuffer",
                "access": "readWrite"
            }
        },
        {
            "name": "i2D",
            "binding": {"kind": "descriptorTableSlot", "index": 1},
            "type": {
                "kind": "resource",
                "baseShape": "texture2D",
                "access": "readWrite"
            }
        }
    ],
    "entryPoints": [
        {
            "name": "main",
            "stage:": "fragment"
        }
    ]
}
}
